src/Uno.UI/Controls/FauxGradientBorderPresenter.cs

```diff
@@ -0,0 +1,135 @@
+﻿#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using Uno.Foundation.Logging;
+using Uno.UI.Xaml.Media;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Media;
+using Uno.UI.Xaml;
+
+namespace Uno.UI.Controls;
+
+/// <summary>
+/// This presenter provides a way to display "fake" LinearGradientBrush on element border for
+/// cases which are unsupported by WebAssembly, iOS and macOS.
+/// </summary>
+/// <remarks>
+/// WASM - the presenter will be used in case the element has CornerRadius applied.
+/// iOS and macOS - the presenter will be used in case the element has LinearGradientBrush with a transform applied.
+/// All other cases - the presenter is not visible.
+/// </remarks>
+public partial class FauxGradientBorderPresenter : ContentPresenter
```

Comment:
Intended to be public?

Reply:
Yes, at least for now - it could be potentially interesting for community toolkit. Also I think that it will help avoid issues when users make a direct copy of our styles as they want to modify them slightly.